### PR TITLE
FM2-229: Update Task resource to support paging

### PR DESCRIPTION
**Pull request by varung-31**
## Description of what I changed

Update Task resource to return paged search results.
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-229

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api-2.2/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirConditionDaoImpl_2_2.java [hunk 1]
```diff
@@ -151,7 +151,7 @@ protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams
 				case FhirConstants.CODED_SEARCH_HANDLER:
 					entry.getValue().forEach(param -> handleCode(criteria, (TokenAndListParam) param.getParam()));
 					break;
-				case FhirConstants.CONDITION_CLINICAL_STATUS_HANDLER:
+				case FhirConstants.STATUS_SEARCH_HANDLER:
```
Comment (corneliouzbett) [inline api-2.2/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirConditionDaoImpl_2_2.java:154]: These changes aren't part of Task resource. You should move to a different commit, PR

File: api-2.2/src/main/java/org/openmrs/module/fhir2/api/impl/FhirConditionServiceImpl_2_2.java [hunk 2]
```diff
@@ -57,7 +57,7 @@ public IBundleProvider searchConditions(ReferenceAndListParam patientParam, Toke
 		SearchParameterMap theParams = new SearchParameterMap()
 		        .addParameter(FhirConstants.PATIENT_REFERENCE_SEARCH_HANDLER, patientParam)
 		        .addParameter(FhirConstants.CODED_SEARCH_HANDLER, code)
-		        .addParameter(FhirConstants.CONDITION_CLINICAL_STATUS_HANDLER, clinicalStatus)
+		        .addParameter(FhirConstants.STATUS_SEARCH_HANDLER, clinicalStatus)
 		        .addParameter(FhirConstants.QUANTITY_SEARCH_HANDLER, onsetAge)
```
Comment (corneliouzbett) [inline api-2.2/src/main/java/org/openmrs/module/fhir2/api/impl/FhirConditionServiceImpl_2_2.java:61]: Ditto

File: api/src/test/java/org/openmrs/module/fhir2/providers/r3/ConditionFhirR3ResourceProviderTest.java [hunk 3]
```diff
@@ -178,36 +178,36 @@ public void searchConditions_shouldReturnConditionReturnedByService() {
...
-
+		
 		TokenAndListParam clinicalList = new TokenAndListParam();
 		clinicalList.addValue(new TokenOrListParam().add(new TokenParam("test clinical")));
-
+		
 		DateRangeParam onsetDate = new DateRangeParam().setLowerBound("lower date").setUpperBound("upper date");
-
+		
 		QuantityAndListParam onsetAge = new QuantityAndListParam();
```
Comment (corneliouzbett) [inline api/src/test/java/org/openmrs/module/fhir2/providers/r3/ConditionFhirR3ResourceProviderTest.java:195]: Also, remove this formatting changes from this PR

File: api/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceProviderTest.java [hunk 4]
```diff
@@ -170,25 +188,24 @@ public void getTaskHistoryByWithWrongId_shouldThrowResourceNotFoundException() {
...
-	//		status.add(statusToken);
-	//
-	//		Bundle results = resourceProvider.searchTasks(null, null, status, null);
-	//
-	//		assertThat(results, notNullValue());
-	//		assertThat(results.getTotal(), equalTo(1));
-	//		assertThat(results.getEntry(), notNullValue());
-	//		assertThat(results.getEntry().get(0).getResource().fhirType(), equalTo("Task"));
-	//		assertThat(results.getEntry().get(0).getResource().getId(), equalTo(TASK_UUID));
-	//	}
```
Comment (corneliouzbett) [inline api/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceProviderTest.java:193]: I guess you forgot about this. You should probably remove

Comment (varung-31) [reply api/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceProviderTest.java:193]: Oh yeah, I think I forgot.

**Comment by varung-31** [comment]
@corneliouzbett  Please take a look!

File: api/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceProviderTest.java (hunk 4 above)
Comment (varung-31) [reply api/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceProviderTest.java:193]: I have added the correct test for this just below it. Hence deleted.
